forge-gui/res/cardsfolder/f/foxfire.txt

```diff
@@ -2,7 +2,9 @@ Name:Foxfire
 ManaCost:2 G
 Types:Instant
 A:SP$ Untap | Cost$ 2 G | ValidTgts$ Creature.attacking | TgtPrompt$ Select target attacking creature | SubAbility$ DBPump | SpellDescription$ Untap target attacking creature. Prevent all combat damage that would be dealt to and dealt by that creature this turn. Draw a card at the beginning of the next turn's upkeep.
-SVar:DBPump:DB$ Pump | Defined$ Targeted | KW$ Prevent all combat damage that would be dealt to and dealt by CARDNAME. | SubAbility$ DelTrigSlowtrip
+SVar:DBPump:DB$ Effect | ReplacementEffects$ RPrevent1,RPrevent2 | RememberObjects$ Targeted | ExileOnMoved$ Battlefield | SubAbility$ DelTrigSlowtrip
+SVar:RPrevent1:Event$ DamageDone | Prevent$ True | IsCombat$ True | ValidSource$ Card.IsRemembered | Description$ Prevent all combat damage that would be dealt to and dealt by that creature this turn.
+SVar:RPrevent2:Event$ DamageDone | Prevent$ True | IsCombat$ True | ValidTarget$ Card.IsRemembered | Description$ Prevent all combat damage that would be dealt to and dealt by that creature this turn. | Secondary$ True
 SVar:DelTrigSlowtrip:DB$ DelayedTrigger | NextTurn$ True | Mode$ Phase | Phase$ Upkeep | ValidPlayer$ Player | Execute$ DrawSlowtrip | TriggerDescription$ Draw a card.
```

Comment:
TriggerDescription (& SpellDescription??) Draw a card at the beginning of the next turn's upkeep. ?

Reply:
Thanks for all those. Looks like we can combine both for best result:

![image](https://user-images.githubusercontent.com/8506892/170839929-2f8f2a30-a0e6-4296-9315-733484d0c7e8.png)

![image](https://user-images.githubusercontent.com/8506892/170839935-012e8bea-6ffc-4446-a323-f664468a1945.png)